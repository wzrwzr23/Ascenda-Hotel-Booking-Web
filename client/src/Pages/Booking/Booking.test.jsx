import React from "react";
import { render, screen, cleanup, fireEvent} from '@testing-library/react';
import { BrowserRouter, Routes } from "react-router-dom";
const { default: Booking } = require("./Booking");
import ReactDOM from 'react-dom'
import renderer from "react-test-renderer"

afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div")
    render(
        <BrowserRouter><Booking></Booking></BrowserRouter>
    , div)
    ReactDOM.unmountComponentAtNode(div)
})

test('renders and displays User Details', () => {
    render(<BrowserRouter><Booking></Booking></BrowserRouter>);
    const linkElement = screen.getByText(/User Details/i);
    expect(linkElement).toBeInTheDocument();
});

it("renders and shows hopepage link", ()=>{
    render(<BrowserRouter><Booking></Booking></BrowserRouter>);
    expect(screen.getByRole('link', {name: /Ascenda Hotel Booking/i, hidden: true})).toBeEnabled();
    // expect(screen.getByText(/See availability/i, {selector: 'button'}));
})

it("renders and shows option Miss", ()=>{
    render(<BrowserRouter><Booking></Booking></BrowserRouter>);
    expect(screen.getByRole('option', {name: /Miss/i, hidden: true})).toBeEnabled();
    // expect(screen.getByText(/See availability/i, {selector: 'button'}));
})

test("renders and checks if submit button enabled", async ()=>{
    render(<BrowserRouter><Booking></Booking></BrowserRouter>);
    expect(await screen.findByRole('button', { name: /Submit/i })).toBeEnabled();
})

describe('Destination search input: Typing', () => {
    test('search box updates on change', () => {
        const {queryByPlaceholderText} = render(<BrowserRouter><Booking/></BrowserRouter>);
        const searchInput = queryByPlaceholderText('MM/YY');
        fireEvent.change(searchInput, {target: {value: "03/22"}});
        // userEvent.type(searchInput, 'test')
        expect(searchInput.value).toBe("03/22");
    });
})

it("matches snapshot", ()=>{ //if you change the code in footer, then this test will fail. if intentional, can update snapshot. if by mistake, undo the code
    const tree = renderer.create(<BrowserRouter><Booking/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})

// let container = null
// beforeEach(()=>{
//     container = document.createElement("div");
//     document.body.appendChild(container);
// })

// afterEach(()=>{
//     unmountComponentAtNode(container);
//     container.remove();
//     container = null;
// })

// it("renders booking data", async () => {
//     const fakeBooking = {
//         salutation: "Miss",
//         firstName:"Namitha",    //test: with missing field
//         lastName: "Justine",    //test:with missing field
//         phoneNumber: "gde3432e2",
//         emailAddress: "namitha@gmail.com", //test: "namitha@gmail": without the .com
//         billingAddress: "25 Root"
//     }
//     // jest.spyOn(global, "fetch").mockImplementation(() =>
        
//     // );

//     // Use the asynchronous version of act to apply resolved promises
//     await act(function Component() {
//         render(<Booking/>, container);
//     });

//     expect(container.querySelector("div").textContent).toBe(fakeBooking.firstName);
//     expect(container.textContent).toBe(fakeBooking.address);
//     expect(container.textContent).toBe(fakeBooking.emailAddress);

//     // remove the mock to ensure tests are completely isolated
//     global.fetch.mockRestore();
// });