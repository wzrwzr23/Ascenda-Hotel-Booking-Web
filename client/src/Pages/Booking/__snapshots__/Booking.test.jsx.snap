// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="navbar"
  >
    <div
      className="navContainer"
      data-testid="navRegister"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "inherit",
            "textDecoration": "none",
          }
        }
      >
        <span
          className="logo"
        >
          Ascenda Hotel Booking
        </span>
      </a>
    </div>
  </div>,
  <div
    className="header"
  >
    <div
      className="headerContainer listMode"
    >
      <div
        className="headerList"
      >
        <div
          className="headerListItem active"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-house "
            data-icon="house"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Home
          </span>
        </div>
        <div
          className="headerListItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-person-booth "
            data-icon="person-booth"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 496C192 504.8 199.3 512 208 512h32C248.8 512 256 504.8 256 496V320H192V496zM544 0h-32v496c0 8.75 7.25 16 16 16h32c8.75 0 16-7.25 16-16V32C576 14.25 561.8 0 544 0zM64 128c26.5 0 48-21.5 48-48S90.5 32 64 32S16 53.5 16 80S37.5 128 64 128zM224 224H173.1L127.9 178.8C115.8 166.6 99.75 160 82.75 160H64C46.88 160 30.75 166.8 18.75 178.8c-12 12.12-18.72 28.22-18.72 45.35L0 480c0 17.75 14.25 32 31.88 32s32-14.25 32-32L64 379.3c.875 .5 1.625 1.375 2.5 1.75L95.63 424V480c0 17.75 14.25 32 32 32c17.62 0 32-14.25 32-32v-56.5c0-9.875-2.375-19.75-6.75-28.62l-41.13-61.25V253l20.88 20.88C141.8 283 153.8 288 166.5 288H224c17.75 0 32-14.25 32-32S241.8 224 224 224zM192 32v160h64V0H224C206.3 0 192 14.25 192 32zM288 32l31.5 223.1l-30.88 154.6C284.3 431.3 301.6 448 320 448c15.25 0 27.99-9.125 32.24-30.38C353.3 434.5 366.9 448 384 448c17.75 0 32-14.25 32-32c0 17.75 14.25 32 32 32s32-14.25 32-32V0h-192V32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Rooms
          </span>
        </div>
        <div
          className="headerListItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle-plus "
            data-icon="circle-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 368C269.3 368 280 357.3 280 344V280H344C357.3 280 368 269.3 368 256C368 242.7 357.3 232 344 232H280V168C280 154.7 269.3 144 256 144C242.7 144 232 154.7 232 168V232H168C154.7 232 144 242.7 144 256C144 269.3 154.7 280 168 280H232V344C232 357.3 242.7 368 256 368z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Services
          </span>
        </div>
        <div
          className="headerListItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-cart-plus "
            data-icon="cart-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM272 180H316V224C316 235 324.1 244 336 244C347 244 356 235 356 224V180H400C411 180 420 171 420 160C420 148.1 411 140 400 140H356V96C356 84.95 347 76 336 76C324.1 76 316 84.95 316 96V140H272C260.1 140 252 148.1 252 160C252 171 260.1 180 272 180zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Shop
          </span>
        </div>
        <div
          className="headerListItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-box-archive "
            data-icon="box-archive"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 432C32 458.5 53.49 480 80 480h352c26.51 0 48-21.49 48-48V160H32V432zM160 236C160 229.4 165.4 224 172 224h168C346.6 224 352 229.4 352 236v8C352 250.6 346.6 256 340 256h-168C165.4 256 160 250.6 160 244V236zM480 32H32C14.31 32 0 46.31 0 64v48C0 120.8 7.188 128 16 128h480C504.8 128 512 120.8 512 112V64C512 46.31 497.7 32 480 32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Gallery
          </span>
        </div>
        <div
          className="headerListItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-phone "
            data-icon="phone"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            Contact
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="booking"
  >
    <div
      className="bookingUserDetails"
    >
      <h1
        className="detailsTitle"
      >
        BOOKING
      </h1>
      <h2
        className="detailsSubTitle"
      >
        User Details
      </h2>
      <form
        className="user"
      >
        <br />
        <br />
        <div
          className="userName"
        >
          <select
            className="salutation"
            id="salutation"
            name="salutation"
          >
            <option
              value="select"
            >
              Select an option
            </option>
            <option
              id="mr"
              value="mr"
            >
              Mr
            </option>
            <option
              value="mrs"
            >
              Mrs
            </option>
            <option
              value="miss"
            >
              Miss
            </option>
          </select>
          <div
            className="firstName"
          >
            First Name
            <span
              className="require"
            >
              *
            </span>
            <input
              className="inputBox"
              id="firstName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="lastName"
          >
            Last Name
            <span
              className="require"
            >
              *
            </span>
            <input
              className="inputBox"
              id="lastName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="phoneNumber"
        >
          Phone Number
          <span
            className="require"
          >
            *
          </span>
          <input
            className="inputBox"
            id="phone"
            onChange={[Function]}
            required={true}
            type="tel"
            value=""
          />
        </div>
        <div
          className="emailAddress"
        >
          Email Address
          <span
            className="require"
          >
            *
          </span>
          <input
            className="inputBox"
            id="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="cardNumber"
        >
          Card Number
          <span
            className="require"
          >
            *
          </span>
          <input
            className="inputBox"
            id="card"
            name="number"
            onChange={[Function]}
            pattern="^[0-9]{4} [0-9]{4} [0-9]{4} [0-9]{4}$"
            placeholder="0000 0000 0000 0000"
            required={true}
            type="tel"
            value=""
          />
        </div>
        <div
          className="cardExpiry"
        >
          Card Expiry
          <span
            className="require"
          >
            *
          </span>
          <input
            className="inputBox"
            id="expiry"
            name="expiry"
            onChange={[Function]}
            pattern="^[0-9]{2}/[0-9]{2}$"
            placeholder="MM/YY"
            required={true}
            type="tel"
            value=""
          />
        </div>
        <div
          className="cardCVC"
        >
          CVV/CVC
          <span
            className="require"
          >
            *
          </span>
          <input
            className="inputBox"
            id="cvc"
            name="cvc"
            onChange={[Function]}
            pattern="^[0-9]{3}$"
            placeholder="CVC/CVV"
            required={true}
            type="tel"
            value=""
          />
        </div>
        <div
          className="specials"
        >
          Special Requests:
          <div
            className="textBox"
          >
            <textarea
              className="Text1"
              cols="40"
              id="special"
              onChange={[Function]}
              rows="5"
              value=""
            />
          </div>
        </div>
        <div
          className="billingAddress"
        >
          Billing Address
          <span
            className="require"
          >
            *
          </span>
          <div
            className="textBox"
          >
            <textarea
              className="Text1"
              cols="40"
              id="billing"
              onChange={[Function]}
              rows="5"
              value=""
            />
          </div>
        </div>
        <input
          className="submitBtn"
          id="submit"
          onClick={[Function]}
          type="submit"
        />
      </form>
    </div>
  </div>,
  <div
    className="footer"
    data-testid="footer"
  >
    <div
      className="fLists"
    >
      <ul
        className="fList"
      >
        <li
          className="fListItem"
        >
          Countries
        </li>
        <li
          className="fListItem"
        >
          Regions
        </li>
        <li
          className="fListItem"
        >
          Cities
        </li>
        <li
          className="fListItem"
        >
          Districts
        </li>
        <li
          className="fListItem"
        >
          Airports
        </li>
        <li
          className="fListItem"
        >
          Hotels
        </li>
      </ul>
      <div
        className="fBox"
      >
        <div
          className="fBoxItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-location-crosshairs "
            data-icon="location-crosshairs"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M176 256C176 211.8 211.8 176 256 176C300.2 176 336 211.8 336 256C336 300.2 300.2 336 256 336C211.8 336 176 300.2 176 256zM256 0C273.7 0 288 14.33 288 32V66.65C368.4 80.14 431.9 143.6 445.3 224H480C497.7 224 512 238.3 512 256C512 273.7 497.7 288 480 288H445.3C431.9 368.4 368.4 431.9 288 445.3V480C288 497.7 273.7 512 256 512C238.3 512 224 497.7 224 480V445.3C143.6 431.9 80.14 368.4 66.65 288H32C14.33 288 0 273.7 0 256C0 238.3 14.33 224 32 224H66.65C80.14 143.6 143.6 80.14 224 66.65V32C224 14.33 238.3 0 256 0zM128 256C128 326.7 185.3 384 256 384C326.7 384 384 326.7 384 256C384 185.3 326.7 128 256 128C185.3 128 128 185.3 128 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            <h2>
              Address
            </h2>
          </span>
          <br />
          <br />
        </div>
        380 Jln Besar, Singapore 209000
        <div
          className="fBoxItem"
          data-testid="phone"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-phone "
            data-icon="phone"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            <h2>
              Phone
            </h2>
          </span>
          <br />
          <br />
        </div>
        +65 88730488
        <div
          className="fBoxItem"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-at "
            data-icon="at"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.8 20.73c-93.45 18.32-168.7 93.66-187 187.1c-27.64 140.9 68.65 266.2 199.1 285.1c19.01 2.888 36.17-12.26 36.17-31.49l.0001-.6631c0-15.74-11.44-28.88-26.84-31.24c-84.35-12.98-149.2-86.13-149.2-174.2c0-102.9 88.61-185.5 193.4-175.4c91.54 8.869 158.6 91.25 158.6 183.2l0 16.16c0 22.09-17.94 40.05-40 40.05s-40.01-17.96-40.01-40.05v-120.1c0-8.847-7.161-16.02-16.01-16.02l-31.98 .0036c-7.299 0-13.2 4.992-15.12 11.68c-24.85-12.15-54.24-16.38-86.06-5.106c-38.75 13.73-68.12 48.91-73.72 89.64c-9.483 69.01 43.81 128 110.9 128c26.44 0 50.43-9.544 69.59-24.88c24 31.3 65.23 48.69 109.4 37.49C465.2 369.3 496 324.1 495.1 277.2V256.3C495.1 107.1 361.2-9.332 207.8 20.73zM239.1 304.3c-26.47 0-48-21.56-48-48.05s21.53-48.05 48-48.05s48 21.56 48 48.05S266.5 304.3 239.1 304.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span>
            <h2>
              Email
            </h2>
          </span>
          <br />
          <br />
        </div>
        customerservice@ascenda.com
      </div>
    </div>
    <div
      className="fText"
    >
      Copyright © 2022 Ascenda Hotel Booking
    </div>
  </div>,
]
`;
